File: Obj-C/AdGem-ObjC/ViewController.m
//
//  ViewController.m
//  AdGem-ObjC
//
//  Created by Matthew Knippen on 2/20/19.
//  Copyright © 2019 AdGem LLC. All rights reserved.
//

#import "ViewController.h"
#import "AppDelegate.h"
@import AdGemSdk;

@interface ViewController ()

@property (weak, nonatomic) IBOutlet UILabel *versionLabel;
@property (weak, nonatomic) IBOutlet UILabel *rewardLabel;

/// Token for the coin-update observer, removed on dealloc.
@property (strong, nonatomic) id coinsObserver;

@end

@implementation ViewController

- (void)dealloc {
    if (_coinsObserver != nil) {
        [[NSNotificationCenter defaultCenter] removeObserver:_coinsObserver];
    }
}

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    self.versionLabel.text = [NSString stringWithFormat:@"iOS - V%@", AdGem.sdkVersion];

    // Refresh the balance whenever a reward is granted, even while this screen
    // is already visible (the reward callback can land after viewDidAppear has
    // already run).
    __weak typeof(self) weakSelf = self;
    self.coinsObserver = [[NSNotificationCenter defaultCenter] addObserverForName:AdGemCoinsUpdatedNotification
                                                                          object:nil
                                                                           queue:[NSOperationQueue mainQueue]
                                                                      usingBlock:^(NSNotification * _Nonnull note) {
        [weakSelf refreshRewardLabel];
    }];
}

- (void)viewDidAppear:(BOOL)animated {
    [super viewDidAppear:animated];
    [self refreshRewardLabel];
}

- (void)refreshRewardLabel {
    NSInteger coins = [NSUserDefaults.standardUserDefaults integerForKey:@"coins"];
    self.rewardLabel.text = [NSString stringWithFormat:@"%ld coins", (long)coins];
}

- (IBAction)showOfferwallTapped:(id)sender {
    [AdGem showOfferwall];
}


@end

File: Swift/AdGemTester/ViewController.swift
//
//  ViewController.swift
//  AdGemTester
//
//  Created by Matthew Knippen on 4/11/18.
//  Copyright © 2018 AdGem. All rights reserved.
//

import UIKit
import AdGemSdk

class ViewController: UIViewController {

    @IBOutlet weak var versionLabel: UILabel!
    @IBOutlet weak var rewardLabel: UILabel!


    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        self.versionLabel.text = "iOS - V\(AdGem.sdkVersion)"

        // Refresh the balance whenever a reward is granted, even while this
        // screen is already visible (the reward callback can land after
        // viewDidAppear has already run).
        NotificationCenter.default.addObserver(
            forName: .adGemCoinsUpdated,
            object: nil,
            queue: .main
        ) { [weak self] _ in
            self?.refreshRewardLabel()
        }
    }

    override func viewDidAppear(_ animated: Bool) {
        super.viewDidAppear(animated)
        refreshRewardLabel()
    }

    private func refreshRewardLabel() {
        let coins = UserDefaults.standard.integer(forKey: "coins")
        self.rewardLabel.text = "\(coins) coins"
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    @IBAction func showOfferwallTapped(_ sender: Any) {
        AdGem.showOfferwall()
    }
}

